server/src/main/java/org/jetbrains/bsp/bazel/server/sync/JUnitTestParser.kt

```diff
@@ -0,0 +1,131 @@
+package org.jetbrains.bsp.bazel.server.sync
+// TODO - rethink the location
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+import ch.epfl.scala.bsp4j.TaskId
+import ch.epfl.scala.bsp4j.TestReport
+import ch.epfl.scala.bsp4j.TestStatus
+import org.jetbrains.bsp.bazel.bazelrunner.BazelProcessResult
+import org.jetbrains.bsp.bazel.logger.BspClientTaskNotifier
+import java.util.*
+import java.util.regex.Pattern
+
+private data class TestOutputLine(
+  val name: String,
+  val passed: Boolean,
+  val message: String,
+  val indent: Int,
+  val taskId: TaskId?
+)
+
+private data class StartedTestingTarget(
+  val uri: String,
+  val taskId: TaskId
+)
+
+class JUnitTestParser(
+  private val bspClientTaskNotifier: BspClientTaskNotifier
+) {
+  fun processTestOutputWithJUnit(testResult: BazelProcessResult) {
+    val startedSuites: Stack<TestOutputLine> = Stack()
+    var currentTestTarget: StartedTestingTarget? = null
+    var previousOutputLine: TestOutputLine? = null
+
+    testResult.stdoutLines.forEach {
```

Comment:
i'd split those nested if-elses with functions

Reply:
split into:
- `processTestingEndingLine()`
- `processPreviousOutputLine()`
- `getCurrentOutputLine()`
- modified version of `beginTesting()`